packages/js/src/utils/requester.ts

```diff
@@ -1,32 +1,19 @@
 import { isNode } from '@silverhand/essentials';
 
-import { LogtoError } from './errors';
+import { LogtoError, LogtoRequestError } from './errors';
 
-interface LogtoErrorResponse {
-  error?: string;
-  error_description?: string;
+interface LogtoRequestErrorBody {
+  code: string;
+  message: string;
+  data: string;
 }
 
-interface LogtoErrorParameters extends LogtoErrorResponse {
-  message?: string;
-}
-
-const getLogtoErrorParametersByResponse = async (
-  response: Response
-): Promise<LogtoErrorParameters> => {
-  const text = await response.text();
+const getLogtoRequestError = async (response: Response): Promise<LogtoRequestError> => {
   try {
-    const data = JSON.parse(text) as LogtoErrorResponse;
-    if (data.error || data.error_description) {
-      return {
-        error: data.error,
-        error_description: data.error_description,
-      };
-    }
-
-    return { message: response.statusText };
+    const { code, message } = await response.json<LogtoRequestErrorBody>();
+    return new LogtoRequestError(code ?? 'request failed', message);
```

Comment:
we can use superstruct to ensure code has value?

Reply:
Refactored, PTAL.